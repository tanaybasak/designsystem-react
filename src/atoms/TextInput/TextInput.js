import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import prefix from '../../settings';

const TextInput = ({ className, ...restProps }) => {
    const [value, setValue] = useState(restProps.value || "");
    const classnames = `${prefix}-form-control ${className}`.trim();


    useEffect(() => {
        if (restProps.value || restProps.value === '') {
            setValue(restProps.value);
        }
    }, [restProps.value]);

    return (
        <>
            <input
                className={classnames}
                type="text"
                {...restProps}
                value={value}
                onChange={event => {
                    setValue(event.currentTarget.value);
                    restProps.onChange(event);
                }}
            />
        </>
    );
}

TextInput.propTypes = {
    /** Additional class name to be given to <input> tag. */
    className: PropTypes.string,
    /** Specifying the <input> tag is disabled or not. */
    disabled: PropTypes.bool,
    /** Unique Id for the <input> tag. */
    id: PropTypes.string,
    /** Placeholder text for the <input> tag. */
    placeholder: PropTypes.string,
    /** Value of the input field. */
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** The default Value of the field on rendering. */
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Event to subscribe when the value of the Input field changes. */
    onChange: PropTypes.func,
    /** Event to subscribe when the Input field is clicked. */
    onClick: PropTypes.func,
    /** Event to subscribe when the Input field is focused. */
    onFocus: PropTypes.func,
    /** Event to subscribe when the Input field is Blurred. */
    onBlur: PropTypes.func
};

TextInput.defaultProps = {
    className: "",
    disabled: false,
    onChange: () => { },
    onClick: () => { },
    onFocus: () => { },
    onBlur: () => { }
};

export default TextInput;