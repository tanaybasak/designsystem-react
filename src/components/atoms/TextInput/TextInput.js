import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { prefix } from '../../../settings';

export default function TextInput({ className, type, ...restProps }) {
    const [value, setValue] = useState(restProps.value || '');

    return (
        <input
            className={`${prefix}-form-control ${className}`}
            type={type}
            {...restProps}
            value={value}
            onChange={event => {
                setValue(event.currentTarget.value);
                restProps.onChange(event);
            }}
        />
    );
};

TextInput.propTypes = {
    className: PropTypes.string,
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    disabled: PropTypes.bool,
    id: PropTypes.string,
    onChange: PropTypes.func,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    placeholder: PropTypes.string,
    type: PropTypes.oneOf([
        'text', 'email', 'password', 'search', 'tel', 'url'
    ]).isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

TextInput.defaultProps = {
    className: null,
    disabled: false,
    onChange: () => { },
    onClick: () => { },
    onFocus: () => { },
    onBlur: () => { },
    type: 'text'
};