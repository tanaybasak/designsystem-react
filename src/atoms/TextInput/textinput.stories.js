import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
//@update-path-build-start
import TextInput from './TextInput';
//@update-path-build-end

storiesOf('TextInput', module)
  .add(
    'default',
    () => (
      <TextInput
        className=""
        disabled={boolean('Disabled', false)}
        onBlur={action('onBlur')}
        onChange={action('onChange')}
        onClick={action('onClick')}
        onFocus={action('onFocus')}
        placeholder={text('Placeholder', 'Placeholder Text')}
        aria-label="text input"
      />
    ),
    {
      info: {
        text: `Description About TextInput Component \n
      
      import { TextInput } from '@patron/patron-react/textinput';`
      }
    }
  )
  .add(
    'with error',
    () => (
      <div className="hcl-form-group">
        <TextInput
          className=""
          disabled={boolean('Disabled', false)}
          onBlur={action('onBlur')}
          onChange={action('onChange')}
          onClick={action('onClick')}
          onFocus={action('onFocus')}
          placeholder={text('Placeholder', 'Placeholder Text')}
          aria-label="text input"
          data-invalid="true"
        />
        <label className="hcl-label" htmlFor="labeltext">
          Label text
        </label>
        <div className="hcl-error-msg">Validation message here</div>
      </div>
    ),
    {
      info: {
        text: `Description About TextInput Component \n
        
        import { TextInput } from '@patron/patron-react/textinput';`
      }
    }
  )
  .add(
    'with helpertext',
    () => (
      <div className="hcl-form-group">
        <TextInput
          className=""
          disabled={boolean('Disabled', false)}
          onBlur={action('onBlur')}
          onChange={action('onChange')}
          onClick={action('onClick')}
          onFocus={action('onFocus')}
          placeholder={text('Placeholder', 'Placeholder Text')}
          aria-label="text input"
        />
        <label className="hcl-label" htmlFor="labeltext">
          Label text
        </label>
        <div className="hcl-helper-text">Optional Helper text goes here</div>
      </div>
    ),
    {
      info: {
        text: `Description About TextInput Component \n
        
        import { TextInput } from '@patron/patron-react/textinput';`
      }
    }
  );
