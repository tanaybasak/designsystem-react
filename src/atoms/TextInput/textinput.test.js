import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

describe('<TextInput> component', () => {
    it('snapshots/renders TextInput correctly', () => {
        const tree = renderer.create(<TextInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('simulate change event in TextInput and is called Once', () => {
        const mockCallBack = jest.fn();
        const wrapper = shallow(<TextInput type="text" onChange={mockCallBack} />);
        wrapper.find('input[type="text"]').simulate('change', { currentTarget: { value: 'Hello' } });
        expect(mockCallBack.mock.calls.length).toEqual(1);
        expect(wrapper.find('input[type="text"]').props().value).toBe('Hello');
    });

    it('simulate click, onfocus event in TextInput and is called Once', () => {
        const mockCallBackclick = jest.fn().mockReturnValue(undefined);
        const mockCallBackfocus = jest.fn().mockReturnValue(undefined);
        const mockCallBackchange = jest.fn().mockReturnValue(undefined);
        const mockCallBackblur = jest.fn().mockReturnValue(undefined);
        const wrapper = shallow(<TextInput type="text" />);
        wrapper.find('input[type="text"]').simulate('click', { target: {} });
        wrapper.find('input[type="text"]').simulate('focus', { target: {} });
        wrapper.find('input[type="text"]').simulate('change', { currentTarget: { value: 'Hello' } });
        wrapper.find('input[type="text"]').simulate('blur', { target: {} });
        expect(mockCallBackclick()).toEqual(undefined);
        expect(mockCallBackfocus()).toEqual(undefined);
        expect(mockCallBackchange()).toEqual(undefined);
        expect(mockCallBackblur()).toEqual(undefined);
    });
});