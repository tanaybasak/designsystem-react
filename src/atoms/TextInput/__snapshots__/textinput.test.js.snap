// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> component snapshots/renders TextInput correctly 1`] = `
<input
  className="hcl-form-control"
  disabled={false}
  onChange={[Function]}
  type="text"
  value=""
/>
`;
